include/mbedtls/base64.h

```diff
@@ -42,7 +42,9 @@ extern "C" {
 /**
  * \brief          Encode a buffer into base64 format
  *
- * \param dst      destination buffer
+ * \param dst      destination buffer (can be NULL for checking size, also src
+ *                 and dst buffers may overlap: int this case inplace encoding
+ *                 takes place and src buffer is overwritten)
```

Comment:
Note that in case of partial overlap, this function is only allowed to write between `dst` and `dst + dlen`. It is not allowed to overwrite any part of the `src` buffer that does not overlap, since `src` is a `const *` parameter.

Reply:
Not sure to understand. The proposed patch is intended to support full buffer overlapping to potentially save memory on some scenario. Former implementation was not allowing this. I don't think partial overlapping is that relevant.
But maybe I missed something there..